src/madness/world/worldrmi.cc

```diff
@@ -332,10 +335,19 @@ namespace madness {
     void RMI::RmiTask::huge_msg_handler(void *buf, size_t /*nbytein*/) {
         const size_t* info = (size_t *)(buf);
         int nword = HEADER_LEN/sizeof(size_t);
-        int src = info[nword];
-        size_t nbyte = info[nword+1];
-
-        RMI::task_ptr->hugeq.push_back(std::make_pair(src,nbyte));
+        const int src = info[nword];
+        const size_t nbyte = info[nword+1];
+        const int tag = info[nword+2];
+
+        // extra dose of paranoia: assert that we never process so many huge messages
+        // that the tag wraparound somewhere becomes possible ...
+        // the worst case is where only one node sends huge messages to every node in the communicator
+        // AND it has enough threads to use up all tags
+        // NB list::size() is O(1) in c++11, but O(N) in older libstdc++
+        MADNESS_ASSERT(ThreadPool::size() < RMI::RmiTask::unique_tag_period() ||
+                       RMI::task_ptr->hugeq.size() <
+                           RMI::RmiTask::unique_tag_period() / comm.Get_size());
```

Comment:
I know this has already been merged, but I think this check needs to be modified.
1. The first check, `ThreadPool::size() < RMI::RmiTask::unique_tag_period()`, is constant throughout the run. If it is true then the second half of the check will do nothing.
2. The number of threads does not affect a possible wraparound error. There is nothing stoping 2048 threads from each sending 2 messages (or other combinations). The thread check does not add anything.
3. The second check will always fail when `comm.Get_size() >= RMI::RmiTask::unique_tag_period()`
4. The check should always be on, not just for debug builds.

I think the simple check is the best choice, just check that `RMI::task_ptr->hugeq.size() <= RMI::RmiTask::unique_tag_period()`.